docs/src/architecture/assets.md

```diff
@@ -0,0 +1,61 @@
+
+* Asset exchange should be parallelizable 
+* Asset ownership should be private 
+* Asset usage should be indeed censorship resistant
+* Fees can be paid using any asset
+
+All native assets in Miden are stored directly in accounts, like Ether in Ethereum. Miden does not track ownership of assets using global hashmaps, e.g., ERC20 contracts. Storage of assets locally in accounts provides privacy and the ability for client-side proofs. That is because ownership changes always only involve one account and not the change of a global hashmap. Thus, they can happen in parallel. Additionally, asset exchange is censorship resistant at this level because there is no global contract the transfer must pass through. Finally, users can pay fees in any asset.
+
+## Native assets
+Native assets are data structures issued by asset accounts and can be stored in account and note vaults. All native assets are encoded using a single `Word` (4 field elements), so it is easy to determine the type of an asset inside and outside Miden VM. The asset encodes both the ID of the issuing account and asset details. So, when we look at an asset, we know who issued it. It is guaranteed that there can never be a collision between a fungible and a non-fungible asset.
+
+### Issuance
+Specialized asset accounts called faucets issue assets by creating notes. Like any other account, faucets have an Account ID called `faucet_id`. Users can define specific properties of the asset when creating the faucet. Asset creation is similar to creating ERC20 contracts on Ethereum. The rules for who can issue assets and how are fully customizable and defined by the code deployed in the faucet account. The only restrictions are:
+
+* max supply for fungible assets, and 
+* uniqueness of non-fungible assets.
+
+Faucets can create assets and immediately distribute them by producing notes. However, assets can also stay in the faucet after creation to be sent later, e.g., in a bundle. That way, one can mint a million NFTs locally in a single transaction and then send them out as needed in separate transactions in the future.
+
+<p align="center">
+    <img src="../diagrams/architecture/asset/Asset_Issuance.png">
+</p>
+
+### Fungible assets
+A fungible asset is encoded using the amount and the `faucet_id` of the faucet which issued the asset. The amount is guaranteed to be `2^63 - 1` or smaller, the maximum supply for any fungible asset. Examples of fungible assets are ETH and stablecoins, e.g., DAI, USDT, and USDC.
+
+If the `faucet_id` of MATIC were to be `2`, 100 MATIC are encoded as `[100, 0, 0, 2]` - whereas the `0`s in the middle help to quickly distinguish between fungible and non-fungible assets. 
+
+Note: It is impossible to find a collision between two fungible assets issued by different faucets as the `faucet_id` is included in the asset's description. The id is guaranteed to be other for each faucet as per the faucet creation logic.
```

Comment:
I would probably remove this paragraph as I'm not sure it adds a lot of value and is not exactly correct as written (i.e., we could have two fungible assets each describing 100 MATIC and they would be identical - technically it is a collision but not relevant for how we use it).

Reply:
I am not sure, if I follow. 

```
It is impossible to find a collision between two fungible assets issued by different faucets
```

That should be correct, no? If the `faucet_id` is different and part of the `Word` that represents the faucet, it should be true.